CIScripts/Test/Tests/Protocol/TunnellingWithAgent.Tests.ps1

```diff
@@ -139,6 +139,117 @@ function Test-MPLSoUDP {
     }
 }
 
+function Start-UDPEchoServerInContainer {
+    Param (
+        [Parameter(Mandatory=$true)] [PSSessionT] $Session,
+        [Parameter(Mandatory=$true)] [String] $ContainerName,
+        [Parameter(Mandatory=$true)] [Int16] $ServerPort,
+        [Parameter(Mandatory=$true)] [Int16] $ClientPort
+    )
+    $UDPEchoServerCommand = ( `
+    '$SendPort = {0};' + `
+    '$RcvPort = {1};' + `
+    '$IPEndpoint = New-Object System.Net.IPEndPoint([IPAddress]::Any, $RcvPort);' + `
+    '$UDPSocket = New-Object System.Net.Sockets.UdpClient($IPEndpoint);' + `
+    '$RemoteIPEndpoint = New-Object System.Net.IPEndPoint([IPAddress]::Any, 0);' + `
+    'while($true) {{' + `
+    '    $Payload = $UDPSocket.Receive([ref]$RemoteIPEndpoint);' + `
+    '    $RemoteIPEndpoint.Port = $SendPort;' + `
+    '    $UDPSocket.Send($Payload, $Payload.Length, $RemoteIPEndpoint);' + `
+    '    \"Received message and sent it to: $RemoteIPEndpoint.\" | Out-String;' + `
+    '}}') -f $ClientPort, $ServerPort
```

Comment:
It would be nice to have a Dockerfile for UDP echo server container, like in TCP test.

Same applies to other UDP related functions.